libs/facilities/frontend/view/src/lib/components/detail/detail.page.ts

```diff
@@ -1,84 +1,205 @@
+                    type: 'line',
+                    data: Flow,
+                    itemStyle: { color: Colors.WATER },
+                    lineStyle: { color: Colors.WATER },
+                },
+                {
+                    name: 'Motor Current (V)',
+                    type: 'line',
+                    itemStyle: { color: Colors.MOTORCURRENT },
+                    lineStyle: { color: Colors.MOTORCURRENT },
+                    data: MotorCurrent,
+                },
+                {
+                    name: 'Stuffing Box Temperature (°C)',
+                    type: 'line',
+                    itemStyle: { color: Colors.TEMPERATURE },
+                    lineStyle: { color: Colors.TEMPERATURE },
+                    data: StuffingBoxTemperature,
+                },
+                {
+                    name: 'Pressure In (hPa)',
+                    type: 'line',
+                    itemStyle: { color: Colors.PRESSURE1 },
+                    lineStyle: { color: Colors.PRESSURE1 },
+                    data: PressureIn,
+                },
+                {
+                    name: 'Pressure Out (hPa)',
+                    type: 'line',
+                    itemStyle: { color: Colors.PRESSURE2 },
+                    lineStyle: { color: Colors.PRESSURE2 },
+                    data: PressureOut,
+                },
+            ],
+
+        };
+    });
+
+    protected readonly envChart: Signal<EChartsOption | undefined> = computed(() => {
+        const envData = this.envData();
+        if (!envData)
+            return undefined;
+
+        const Temperature = envData.map((item) => [ item.time, item['Temperature'] ]);
+        const Humidity = envData.map((item) => [ item.time, item['Humidity'] ]);
+        const Pressure = envData.map((item) => [ item.time, item['Pressure'] ]);
+
+        return {
+            ...this.defaultOptions,
+            title: {
+                text: 'Environment Data',
+                left: 'center',
+            },
+            series: [
+                {
+                    name: 'Temperature (°C)',
+                    type: 'line',
+                    itemStyle: { color: Colors.TEMPERATURE },
+                    lineStyle: { color: Colors.TEMPERATURE },
+                    data: Temperature,
+                },
+                {
+                    name: 'Humidity (%)',
+                    type: 'line',
+                    itemStyle: { color: Colors.WATER },
+                    lineStyle: { color: Colors.WATER },
+                    data: Humidity,
+                },
+                {
+                    name: 'Pressure (kPa)',
+                    type: 'line',
+                    itemStyle: { color: Colors.PRESSURE1 },
+                    lineStyle: { color: Colors.PRESSURE1 },
+                    data: Pressure,
+                },
+            ],
+
+        };
+    });
+
+    protected locked$ = new BehaviorSubject<boolean>(true);
```

Comment:
Because I added the locked$ like 4 weeks ago when I did not know about signals. We also already talked about it like 3-4 times and I added the $ in the front at first and then in the back because of it.

I know did it as a signal (but I could not test it yet so I am a little uncertain)

Reply:
true 👍